src/Security/Authentication/Negotiate/src/NegotiateHandler.cs

```diff
@@ -82,6 +82,7 @@ public async Task<bool> HandleRequestAsync()
                 {
                     // HTTP/1.0 and HTTP/1.1 are supported. Do not throw because this may be running on a server that supports
                     // additional protocols.
+                    Logger.ProtocolNotSupported(Request.Protocol);
```

Comment:
HandleRequest is called on every request, even if Authenticate isn't called. Move this to HandleAuthenticateAsync or HandleChallengeAsync.